### PR TITLE
add pipeline message format type

### PR DESCRIPTION
**Pull request by zhenglaizhang**

### CODE REVIEW

**Comment by PhoenixHe-NV** [comment]
How about enable monorepo in this repo? https://rushjs.io/pages/intro/get_started/

File: src/tests/types/message.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      "level": "Error",
+      "message": "the tool failed",
+      "time": "2012-04-23T18:25:43.511Z",
+      "extra": {
+        "role": "sdk automation"
+      }
+    }
+  ]
+}`;
+
```
Comment (ruowan) [inline src/tests/types/message.test.ts]: How about store JSON as file not in source code?

Comment (zhenglaizhang) [reply src/tests/types/message.test.ts]: thanks @ruowan  good suggestion, have updated, plz help review again.

**Comment by zhenglaizhang** [comment]
@NullMDR yes, will try this.

File: src/types/message.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+export enum MessageType {
+  Result = "Result",
+  Raw = "Raw"
+}
+
+export enum MessageLevel {
+  Info = "Info",
```
Comment (ruowan) [inline src/types/message.ts]: You used string enum here. How about union type?  For string enum, I prefer union type.

`MessageLevel = "Info"|"Error"`

File: src/types/message.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  Warning = "Warning",
+  Error = "Error"
+}
+
+export type JsonPath = {
```
Comment (ruowan) [inline src/types/message.ts:3]: What's the tag used for?

File: src/types/message.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export type Extra = {
+  [key: string]: any;
+};
+
+export type BaseMessageRecord = {
+  level: MessageLevel;
+  message: string;
+  time: Date;
```
Comment (ruowan) [inline src/types/message.ts:15]: What's the time used for?  To log execution time?

File: src/types/message.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+export type ResultMessageRecord = BaseMessageRecord & {
+  id?: string;
+  code?: string;
+  docUrl?: string;
+  paths: JsonPath[];
+};
+
+export type RawMessageRecord = BaseMessageRecord & {};
```
Comment (ruowan) [inline src/types/message.ts]: Why "& {}"?

File: src/types/message.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  type: MessageType;
+  data: RawMessageRecord[];
+};
+
+export type ResultMessage = {
+  type: MessageType;
+  data: ResultMessage[];
+};
+
+export type Message = ResultMessage | RawMessage;
```
Comment (ruowan) [inline src/types/message.ts:38]: You may need to provide a type guard here. To check whether type is resultMessage or RawMessage. 

This [doc](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types)

File: src/types/message.ts (hunk 2 above)
Comment (zhenglaizhang) [reply src/types/message.ts]: yes, agreed with you, updated as unions

File: src/types/message.ts (hunk 4 above)
Comment (zhenglaizhang) [reply src/types/message.ts:15]: yes

File: src/types/message.ts (hunk 5 above)
Comment (zhenglaizhang) [reply src/types/message.ts]: as placeholder, removed.

File: src/types/message.ts (hunk 6 above)
Comment (zhenglaizhang) [reply src/types/message.ts:38]: agreed, updated

File: package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  "scripts": {
+    "test": "jest"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/ts-common/specs-scripts-common.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (ruowan) [inline package.json:15]: Why ISC here?

**Review by ruowan — APPROVED** [review]

**Comment by zhenglaizhang** [comment]
@NullMDR @ruowan Sorry that I will close this PR, as I will send out another PR, as that would be based on rush monorepo setup

**Pull request closed without merging**